service/src/test/java/uk/gov/hmcts/reform/fpl/model/CaseDataTest.java

```diff
@@ -1384,6 +1385,125 @@ void shouldReturnAnEmptyOptionalWhenExpectedOrderIsNotContainedWithinHearingOrde
         }
     }
 
+    @Nested
+    class BuildRespondentStatementDynamicList {
+        @Test
+        void shouldBuildDynamicRespondentStatementListFromRespondents() {
+            List<Element<Respondent>> respondents = createRespondents();
+
+            CaseData caseData = CaseData.builder().respondents1(respondents).build();
+            IncrementalInteger i = new IncrementalInteger(1);
+
+            DynamicList expectedDynamicList = ElementUtils
```

Comment:
minor: not big fan of this, as test code uses same logic as production code so wont be able to spot issues, there is DynamicListHelper class that may be used